File: 04_move_zeros.c
void pushZerosToEnd(int arr[], int n) {
    int non_zero = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) {
            if (i != non_zero) {
                int temp = arr[i];
                arr[i] = arr[non_zero];
                arr[non_zero] = temp;
            }
            non_zero++;
        }
    }
}

File: 04_move_zeros.py
class Solution:
    def pushZerosToEnd(self, arr):
        non_zero = 0
        n = len(arr)
        for i in range(n):
            if arr[i] != 0:
                if i != non_zero:
                    arr[i], arr[non_zero] = arr[non_zero], arr[i]
                non_zero += 1